content/docs/thinking-in-react.md

```diff
@@ -31,116 +31,116 @@ Our JSON API returns some data that looks like this:
 
-So finally, our state is:
+आखिरकार हमारा state होगा:
 
-  * The search text the user has entered
-  * The value of the checkbox
+  * उपभोक्ता ने जो सर्च टेक्स्ट डाला है 
+  * चेकबॉक्स की वैल्यू
 
-## Step 4: Identify Where Your State Should Live {#step-4-identify-where-your-state-should-live}
+## स्टेप 4: जानिए आपके state का निवास {#step-4-identify-where-your-state-should-live}
 
-<p data-height="600" data-theme-id="0" data-slug-hash="qPrNQZ" data-default-tab="js" data-user="lacker" data-embed-version="2" class="codepen">See the Pen <a href="https://codepen.io/gaearon/pen/qPrNQZ">Thinking In React: Step 4</a> on <a href="https://codepen.io">CodePen</a>.</p>
+<p data-height="600" data-theme-id="0" data-slug-hash="qPrNQZ" data-default-tab="js" data-user="lacker" data-embed-version="2" class="codepen">पेन को देखे <a href="https://codepen.io/gaearon/pen/qPrNQZ">React में सोचना: स्टेप 4</a>, <a href="https://codepen.io">कोडपेन </a>पर।</p>
 
-OK, so we've identified what the minimal set of app state is. Next, we need to identify which component mutates, or *owns*, this state.
+ठीक, तो हमने एप्प state का न्यूतम सेट पहचान लिया है। अब, हमे ये पहचानना है की इस state का कोनसा कौम्पोनॅन्ट म्यूटेट होता है या *कोन मालिक है*।
 
-Remember: React is all about one-way data flow down the component hierarchy. It may not be immediately clear which component should own what state. **This is often the most challenging part for newcomers to understand,** so follow these steps to figure it out:
+याद रखे: React कौम्पोनॅन्ट हायरार्की के नीचे के बहाव में one-way डाटा के बारे में है। यह शायद एकदम से साफ नहीं होगा की कोनसा कौम्पोनॅन्ट state का मालिक है। **अक्सर यह सबसे ज्यादा चुनौतीपूर्ण भाग होता है अपरिचित लोगो के लिए समझने के लिए**, तो समझने के लिए यह कीजिये: 
 
-For each piece of state in your application:
+अपनी एप्प के हर state के भाग के लिए: 
 
-  * Identify every component that renders something based on that state.
-  * Find a common owner component (a single component above all the components that need the state in the hierarchy).
-  * Either the common owner or another component higher up in the hierarchy should own the state.
-  * If you can't find a component where it makes sense to own the state, create a new component solely for holding the state and add it somewhere in the hierarchy above the common owner component.
+  * हर उस कौम्पोनॅन्ट को पहचाने जो state के आधार पर कुछ रेंडर करता है।
+  * एक कॉमन मुख्य कौम्पोनॅन्ट पहचाने (एकलौता कौम्पोनॅन्ट, सारे कौम्पोनॅन्टस के ऊपर, जिसे state की हायरार्की चाहिए होगी।
+  * या तो कॉमन मुख्या या तो दूसरा कौम्पोनॅन्ट, जो हायरार्की में ऊपर है, state का मालिक होना चाहिए।
+  * यदि आप वह कौम्पोनॅन्ट नहीं पहचान पा रहे जो state का मालिक होना चाहिए, एक नया कौम्पोनॅन्ट बनाइये जो सिर्फ state रखेगा ओर उसे हायरार्की में कही कॉमन मुख्य कौम्पोनॅन्ट के ऊपर रख दीजिये।
```

Comment:
`ओर` => `और`

Reply:
corrected at few other places also, please check the last commit